src/structures/Interaction.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Base = require('./Base');
+const Snowflake = require('../util/Snowflake');
+
+/**
+ * Represents an interaction, see {@link InteractionClient}.
+ * @extends {Base}
+ */
+class Interaction extends Base {
+  constructor(client, data, syncHandle) {
+    super(client);
+    this.syncHandle = syncHandle;
+    this._patch(data);
+  }
+
+  _patch(data) {
+    /**
+     * The ID of this interaction.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.id = data.id;
+
+    /**
+     * The token of this interaction.
+     * @type {string}
+     * @readonly
+     */
+    this.token = data.token;
+
+    /**
+     * The ID of the invoked command.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.commandID = data.data.id;
```

Comment:
this may be brittle logic as new interaction types are released. the format of `data` is dependent on `type`